selfdrive/loggerd/logger.cc

```diff
@@ -132,8 +132,22 @@ static LoggerHandle* logger_open(LoggerState *s, const char* root_path) {
   return h;
 fail:
   LOGE("logger failed to open files");
-  if (h->qlog_file) fclose(h->qlog_file);
-  if (h->log_file) fclose(h->log_file);
+  if (h->qlog_file) {
+    fclose(h->qlog_file);
+    h->qlog_file = NULL;
+  }
+  if (h->log_file) {
+    fclose(h->log_file);
+    h->log_file = NULL;
+  }
+  if (h->bz_file) {
+    BZ2_bzWriteClose(&bzerror, h->bz_file, 0, NULL, NULL);
```

Comment:
Shouldn't the fcloses be moved to after the bzwrite? Otherwise it will write to a closed file?

Reply:
yes,my fault